nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/main/java/org/apache/nifi/lookup/RestLookupService.java

```diff
@@ -320,13 +335,20 @@ public Optional<Record> lookup(Map<String, Object> coordinates, Map<String, Stri
         Request request = buildRequest(mimeType, method, body, endpoint, context);
         try {
             Response response = executeRequest(request);
+            final ResponseBody responseBody = response.body();
 
             if (getLogger().isDebugEnabled()) {
                 getLogger().debug("Response code {} was returned for coordinate {}",
                         new Object[]{response.code(), coordinates});
             }
 
-            final ResponseBody responseBody = response.body();
+            if (!response.isSuccessful()
+                    && ResponseHandlingStrategy.EVALUATED.equals(responseHandler.asAllowableValue(ResponseHandlingStrategy.class))) {
+                final String responseText = responseBody == null ? "<No Message Received from Server>" : responseBody.string();
```

Comment:
Recommend replacing  angle brackets with square brackets in the placeholder message:
```suggestion
                final String responseText = responseBody == null ? "[No Message Received]" : responseBody.string();
```